sinatra/helpers/application_helpers.rb

```
@@ -0,0 +1,114 @@
+          settings.forecast_cache_expiration = Time.now + 3600 # 1 hour cache
+          logger.info "Weather forecast cache updated."
+          return new_forecast # Return the newly fetched data
+        else
+          logger.warn "Failed to fetch new forecast data. Returning error or potentially stale data if available."
+          return new_forecast # Return the error message from fetch_forecast
+        end
+      else
+        logger.debug "Weather forecast cache hit." # Use debug for less noise
+        return cache # Return data from cache
+      end
+    end
```

RuboCop (1.73)

[convention] Assignment Branch Condition size for get_cached_forecast is too high. [<5, 22, 6> 23.35/23]

(Metrics/AbcSize)